core/dbt/parser/schemas.py

```diff
@@ -927,6 +929,28 @@ def _validate_constraint_prerequisites(self, model_node: ModelNode):
         if str(model_node.language) != "sql":
             errors.append(f"Language Error: Expected 'sql' but found '{model_node.language}'")
 
+        # check for primary key constraints defined at the column level
+        pk_col: List[str] = []
+        for col in model_node.columns.values():
+            for constraint in col.constraints:
+                if constraint.type == "primary_key":
```

Comment:
nit: this could be `constraint.type == ConstraintType.primary_key` instead of a hard-coded string.

Reply:
ditto for the comparisons below